### PR TITLE
[Merged by Bors] - feat(Algebra/Associated): add `Associated.map`

### PR DESCRIPTION
**Pull request by acmepjz**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Associated.lean [hunk 1]
```diff
@@ -420,6 +420,11 @@ protected def setoid (α : Type*) [Monoid α] :
   iseqv := ⟨Associated.refl, Associated.symm, Associated.trans⟩
 #align associated.setoid Associated.setoid
 
+theorem map {M N : Type*} [Monoid M] [Monoid N] (f : M →* N) {x y : M}
```
Comment (riccardobrasca) [inline Mathlib/Algebra/Associated.lean:423]: Can you write it using `MonoidHomClass`? Thanks!

**Review by riccardobrasca — APPROVED** [review]
Thanks!

bors d+

File: Mathlib/Algebra/Associated.lean (hunk 1 above)
Comment (acmepjz) [reply Mathlib/Algebra/Associated.lean]: Done. Please check that if the function prototype is correct.

Comment (riccardobrasca) [reply Mathlib/Algebra/Associated.lean]: LGTM, let's just wait for the build to finish.

**Pull request closed without merging**